src/s3/util/s3Tab.ts

```diff
@@ -4,100 +4,88 @@
+        await this.openFile(this.fileUri, workspace)
     }
 
-    public async openFileInEditMode(workspace = vscode.workspace): Promise<vscode.TextEditor | undefined> {
+    public async openFile(uri: vscode.Uri, workspace = vscode.workspace): Promise<void> {
         //find if there is any active editor for this uri
-        const openEditor = await this.getActiveEditor()
-
-        if (!openEditor) {
-            //there wasn't any open, just display it regularly
-            const doc = await workspace.openTextDocument(this.fileUri)
-            this.editor = await this.window.showTextDocument(doc, { preview: false })
-
-            return this.editor
-        } else if (openEditor.document.uri.scheme === 'file') {
-            //there is a tab for this uri open in edit-mode, just shift focus to it by reopening it with the ViewColumn option
-            await this.window.showTextDocument(openEditor.document, {
-                preview: false,
-                viewColumn: openEditor.viewColumn,
-            })
-            this.editor = openEditor
-
-            return this.editor
-        } else if (openEditor.document.uri.scheme === 's3') {
-            // there is a read-only tab open, it needs to be focused, then closed
-            await this.window.showTextDocument(openEditor.document, {
-                preview: false,
-                viewColumn: openEditor.viewColumn,
-            })
-            await vscode.commands.executeCommand('workbench.action.closeActiveEditor')
-            //good to open in edit-mode
-            const doc = await workspace.openTextDocument(this.fileUri)
-            this.editor = await this.window.showTextDocument(doc, { preview: false })
-
-            return this.editor
+        const openEditor = this.editor
+        try {
+            const doc = await workspace.openTextDocument(uri)
+            if (!openEditor) {
+                //there wasn't any open, just display it regularly
+                this.editor = await this.window.showTextDocument(doc, { preview: false })
+            } else if (openEditor.document.uri.scheme === 'file' || openEditor.document.uri.scheme === uri.scheme) {
+                //there is a tab for this uri scheme open, just shift focus to it by reopening it with the ViewColumn option
+                await this.window.showTextDocument(openEditor.document, {
+                    preview: false,
+                    viewColumn: openEditor.viewColumn,
+                })
+                this.editor = openEditor
+            } else {
+                // there is already a tab open, it needs to be focused, then closed
+                await this.focusAndCloseTab()
+                //good to open in given mode
+                this.editor = await this.window.showTextDocument(doc, { preview: false })
+            }
+        } catch (e) {
+            this.window.showErrorMessage('Error opening file ', e)
```

Comment:
Wouldn't this show the full stack trace of the error? It's usually better to hand-craft an error message to display to the user and then log the full error.